server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -2245,29 +2376,41 @@ private void insertIntoUpgradeSegmentsTable(
     }
   }
 
-  private List<DataSegment> retrieveSegmentsById(Handle handle, String datasource, Set<String> segmentIds)
+  private List<DataSegmentWithSchemaInformation> retrieveSegmentsById(Handle handle, String datasource, Set<String> segmentIds)
   {
     if (segmentIds.isEmpty()) {
       return Collections.emptyList();
     }
 
     return SqlSegmentsMetadataQuery.forHandle(handle, connector, dbTables, jsonMapper)
-                                   .retrieveSegmentsById(datasource, segmentIds)
+                                   .retrieveSegmentsById(datasource, segmentIds, centralizedDatasourceSchemaConfig.isEnabled())
                                    .stream()
                                    .map(DataSegmentPlus::getDataSegment)
+        .map(v -> new DataSegmentWithSchemaInformation(v, null, null))
```

Comment:
Why are the numRows and schemaId always null here?

Reply:
Thanks for pointing it out. It is a bug, probably got introduced while merging with master.